kythe/go/platform/kzip/info/info.go

```diff
@@ -43,10 +46,30 @@ func KzipInfo(f kzip.File, scanOpts ...kzip.ScanOption) (*apb.KzipInfo, error) {
 
 	err := kzip.Scan(f, func(rd *kzip.Reader, u *kzip.Unit) error {
 		kzipInfo.TotalUnits++
-		corpusInfo(u.Proto.GetVName().GetCorpus()).Units[u.Proto.GetVName().GetLanguage()]++
+
+		// The unit VName generally does not specify a corpus, so we have to
+		// determine corpus from the source files.
+		var srcCorpora stringset.Set
+		srcs := stringset.New(u.Proto.SourceFile...)
+
 		for _, ri := range u.Proto.RequiredInput {
 			kzipInfo.TotalFiles++
-			corpusInfo(ri.GetVName().GetCorpus()).Files[ri.GetVName().GetLanguage()]++
+			// File VNames don't generally specify a language, so we determine
```

Comment:
File VNames are actually forbidden from specifying a language.

Reply:
The VNames in the graph for `file` nodes mustn't have a language, but there's no prohibition of the VName for the required input having no language.  It could potentially be useful to specify in cross-language compilations.

Reply:
Given the structure and format of a CompilationUnit, I don't think cross-language compilations in the same CU are viable.  As such, I think the risk of accidentally producing invalid file VNames in the graph abides; required_input VNames should not have a language.

Reply:
We already have "cross-language" compilations in the form of JVM-interop compilations (Java, Kotlin, etc.) where the required input class files may be from different source languages.  It may be useful to tag those for a consuming indexer.  The language VName field seems like a potentially appropriate location.  We could instead use a details message, though.

Reply:
I updated the comment. Is there anything else to do here for this PR? Sounds like recording a language per-file could be useful for cross-language cases and this function should be updated if/when we do that.